packages/@aws-cdk/aws-codedeploy/lib/deployment-group.ts

```diff
@@ -141,19 +172,84 @@ export class ServerDeploymentGroup extends ServerDeploymentGroupRef {
             assumedBy: new cdk.ServicePrincipal('codedeploy.amazonaws.com'),
             managedPolicyArns: ['arn:aws:iam::aws:policy/service-role/AWSCodeDeployRole'],
         });
+        this._autoScalingGroups = (props && props.autoScalingGroups) || [];
+        this.addCodeDeployAgentInstallUserData(props && props.agentInstallationType);
 
         const resource = new cloudformation.DeploymentGroupResource(this, 'Resource', {
             applicationName: this.application.applicationName,
             deploymentGroupName: props && props.deploymentGroupName,
             serviceRoleArn: this.role.roleArn,
             deploymentConfigName: props && props.deploymentConfig &&
                 props.deploymentConfig.deploymentConfigName,
+            autoScalingGroups: props && props.autoScalingGroups &&
+                props.autoScalingGroups.map(asg => asg.autoScalingGroupName()),
         });
 
         this.deploymentGroupName = resource.ref;
         this.deploymentGroupArn = deploymentGroupName2Arn(this.application.applicationName,
             this.deploymentGroupName);
     }
+
+    public get autoScalingGroups(): autoscaling.AutoScalingGroup[] {
```

Comment:
What's the rational for cloning?

Reply:
Arrays are mutable, so we defensively copy them in accessors. We do it in a ton of places already (Pipeline, for example).

Reply:
Why would someone need to be able to access the list of ASGs?

Reply:
We will need it in the CodeDeploy Pipeline Action (we need to permissions to each ASG to read from the Pipeline Bucket).